docs/src/test/scala/docs/http/scaladsl/server/directives/WebSocketDirectivesExamplesSpec.scala

```diff
@@ -107,6 +107,119 @@ class WebSocketDirectivesExamplesSpec extends RoutingSpec {
+      Flow[StrictMessage].map {
+        case TextMessage.Strict(text)   => TextMessage("Hello " + text + "!")
+        case BinaryMessage.Strict(data) => BinaryMessage(Source.single(ByteString("Binary message received: ") ++ data))
+      }
+    val timeout: FiniteDuration = 1.second
+    val maxBytes: Long = 1000
+    val websocketRoute =
+      path("greeter") {
+        handleWsMessages { builder =>
+          builder.toStrictTimeout(timeout).maxStrictSize(maxBytes).handleWith(greeter)
+        }
+      }
+
+    // tests:
+    // create a testing probe representing the client-side
+    val wsClient = WSProbe()
+
+    // WS creates a WebSocket request for testing
+    WS("/greeter", wsClient.flow) ~> websocketRoute ~>
+      check {
+        // check response for WS Upgrade headers
+        isWebSocketUpgrade shouldEqual true
+
+        wsClient.sendMessage(TextMessage(Source.single("Peter"))) // non-strict message
+        wsClient.expectMessage("Hello Peter!")
+
+        wsClient.sendMessage(BinaryMessage(Source.single(ByteString("abcdef")))) // non-strict message
+        wsClient.expectMessage(ByteString("Binary message received: abcdef"))
+
+        wsClient.sendMessage("John")
+        wsClient.expectMessage("Hello John!")
+
+        wsClient.sendCompletion()
+        wsClient.expectCompletion()
+      }
+    //#handle-messages-builder
+  }
+
+  "handle-only-text-messages-builder" in {
+    //#handle-only-text-messages-builder
+    def greeter: Flow[StrictMessage, Message, Any] =
+      Flow[StrictMessage].map {
+        case TextMessage.Strict(text) => TextMessage("Hello " + text + "!")
+      }
+    val timeout: FiniteDuration = 1.second
+    val websocketRoute =
+      path("greeter") {
+        handleWsMessages { builder =>
+          builder.only[TextMessage].toStrictTimeout(timeout).handleWith(greeter)
+        }
+      }
+
+    // tests:
+    // create a testing probe representing the client-side
+    val wsClient = WSProbe()
+
+    // WS creates a WebSocket request for testing
+    WS("/greeter", wsClient.flow) ~> websocketRoute ~>
+      check {
+        // check response for WS Upgrade headers
+        isWebSocketUpgrade shouldEqual true
+
+        wsClient.sendMessage(TextMessage(Source.single("Peter"))) // non-strict message
+        wsClient.expectMessage("Hello Peter!")
+
+        wsClient.sendMessage("John")
+        wsClient.expectMessage("Hello John!")
+
+        wsClient.sendCompletion()
+        wsClient.expectCompletion()
+      }
+    //#handle-only-text-messages-builder
+  }
+
+  "handle-only-binary-messages-builder" in {
+    //#handle-only-binary-messages-builder
+    def greeter: Flow[StrictMessage, Message, Any] =
+      Flow[StrictMessage].map {
+        case BinaryMessage.Strict(data) => BinaryMessage(ByteString("Binary message received: ") ++ data)
+      }
+    val timeout: FiniteDuration = 1.second
+    val websocketRoute =
+      path("greeter") {
+        handleWsMessages { builder =>
```

Comment:
I don't think a `Directive1` makes sense for the builder, because it is not composable with other directives which this implies, better with just a method that returns the builder, and the builder then returns a `Route` which is the end-station, so to speak, of the routing tree.